components/org.wso2.carbon.identity.conditional.auth.typingdna.api/src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/api/core/TypingDNAServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.MalformedInputException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.net.ssl.HttpsURLConnection;
+
+public class TypingDNAServiceImpl {
+
+    private static final Log log = LogFactory.getLog(TypingDNAServiceImpl.class);
+
+    /**
+     * This function is used to delete users' typing patterns that are
+     * saved in typingDNA.
+     *
+     * @throws IOException When error/mistakes in TypingDNA configuration for tenant.
+     */
+    public void deleteTypingPatterns() throws IOException {
+
+        // Getting username & tenant domain & generating userID.
+        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        String tenantDomain = "carbon.super";
+        if (IdentityUtil.threadLocalProperties.get().get("TenantNameFromContext") != null) {
+            tenantDomain = (String) IdentityUtil.threadLocalProperties.get().get("TenantNameFromContext");
+        }
+        String userID = getUserID(username, tenantDomain);
+
+        try {
+
+            String APIKey = CommonUtils.getConnectorConfig("adaptive_authentication.tdna.username", tenantDomain);
+            String APISecret = CommonUtils.getConnectorConfig("__secret__adaptive_authentication.tdna.password", tenantDomain);
+            String region = CommonUtils.getConnectorConfig("adaptive_authentication.tdna.apiregion", tenantDomain);
+
+            String baseurl = buildURL(region, userID);
+            String Authorization = Base64.getEncoder().encodeToString((APIKey + ":" + APISecret).getBytes(StandardCharsets.UTF_8));
+            URL url = new URL(baseurl);
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestMethod("DELETE");
+            connection.setConnectTimeout(5000);
+            connection.setReadTimeout(5000);
+            connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+            connection.setRequestProperty("Authorization", "Basic " + Authorization);
+
+            connection.setUseCaches(false);
+            connection.setDoOutput(true);
+
+            InputStream is = connection.getInputStream();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+            StringBuilder res = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                res.append(line);
+                res.append('\r');
+            }
+            rd.close();
+        } catch (MalformedInputException e) {
+            log.error(e);
+        } catch (IdentityEventException e) {
+            log.error(e);
```

Comment:
Need to throw a proper exception from here and handle it in the places that use this method.